osquery/tables/networking/windows/connectivity.cpp

```diff
@@ -0,0 +1,68 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <string>
+
+#include <netlistmgr.h>
+#include <windows.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genConnectivity(QueryContext& context) {
+  QueryData results;
+  Row r;
+
+  INetworkListManager* mgr = nullptr;
+  HRESULT res = CoCreateInstance(CLSID_NetworkListManager,
+                                 NULL,
+                                 CLSCTX_ALL,
+                                 IID_INetworkListManager,
+                                 reinterpret_cast<void**>(&mgr));
+
+  if (res != S_OK) {
+    TLOG << "Failed to instantiate INetworkListManager";
+    return results;
+  }
+
+  NLM_CONNECTIVITY connectivity;
+  res = mgr->GetConnectivity(&connectivity);
```

Comment:
Should we be paranoid and check that `mgr` is not `nullptr`?

Reply:
That *shouldn't* be necessary here -- according to the `CoCreateInstance` docs, checking for `S_OK` should be enough to ensure that `mgr` is valid.